docs/eample-findBy*.md

```diff
@@ -0,0 +1,142 @@
+
+  submitButton.addEventListener('click', () => {
+    const userInput = document.getElementById('username')
+    const passwordInput = document.getElementById('password')
+    var userName = userInput.value
+    var password = passwordInput.value
+
+    if (!userName) {
+      document.getElementById('username_required').style.display = 'inline'
+    }
+
+    if (!password) {
+      document.getElementById('password_required').style.display = 'inline'
+    }
+
+    if (userName && userName !== 'Bob') {
+      setTimeout(() => {
+        document.getElementById('invalid_password').style.display = 'inline'
+      }, 700)
+    }
+
+    if (password && password !== 'theBuilder') {
+      setTimeout(() => {
+        document.getElementById('invalid_password').style.display = 'inline'
+      }, 700)
+    }
+  })
+
+  return el
+}
+
+describe('findBy* Examples', () => {
+  let div
+  let container
+
+  beforeEach(() => {
+    div = document.createElement('div')
+    container = renderContent(div)
+  })
+
+  it('should show a required field warning for each empty input field', async () => {
+    expect(
+      await findByText(container, (content, element) => {
+        return content !== '' && element.textContent.startsWith('User Name')
+      })
+    ).toBeVisible()
+    expect(
+      await findByText(container, (content, element) => {
+        return content !== '' && element.textContent.startsWith('Password')
+      })
+    ).toBeVisible()
+
+    userEvent.click(
+      getByRole(container, 'button', {
+        name: 'Login',
+      })
+    )
+
+    expect(
+      await findByText(container, (content, element) => {
+        return (
+          content !== '' && element.textContent.includes('User Name required')
+        )
+      })
+    ).toBeVisible()
+    expect(
+      await findByText(container, (content, element) => {
+        return (
+          content !== '' && element.textContent.includes('Password required')
+        )
+      })
+    ).toBeVisible()
+  })
+
+  it('should show an async invalid password error', async () => {
+    const userNameField = getByPlaceholderText(container, 'Enter user name')
+    const passwordField = getByPlaceholderText(container, 'Enter password')
+
+    userEvent.type(userNameField, 'wrongName')
+    userEvent.type(passwordField, 'wrongPassword')
+    // fireEvent.focus(passwordField);
+    // fireEvent.change(passwordField, {
+    //   target: {
+    //     value: "wrong"
+    //   }
+    // });
+
+    // fireEvent.click(
+    //   getByRole(container, "button", {
+    //     name: "Login"
+    //   })
+    // );
+    userEvent.click(
+      getByRole(container, 'button', {
+        name: 'Login',
+      })
+    )
+    // await waitFor(() => {
+    //   expect(getByText(container, "Invalid Password")).toBeVisible()
+    // })
+    // screen.debug(await findByText(container, "Invalid Password"));
+    expect(await findByText(container, 'Invalid User Name')).toBeVisible()
```

Comment:
I don't think `findByText` excludes hidden elements by default, does it? That's why this assertion fails.

Reply:
Ah it turns out it was my own mistake. The test wasn't finding the input element at all, I should have been using `el.querySelector` instead of `document.getElementById` 😅 . The tests pass as expected now!